cmd/kubeadm/app/util/apiclient/idempotency.go

```diff
@@ -55,14 +55,13 @@ type kubernetesObject interface {
 
 // CreateOrUpdate creates a runtime object if the target resource doesn't exist.
 // If the resource exists already, this function will update the resource instead.
-func CreateOrUpdate[T kubernetesObject](client kubernetesInterface[T], obj T) error {
+func CreateOrUpdate[T kubernetesObject](ctx context.Context, client kubernetesInterface[T], obj T) error {
 	var lastError error
-	err := wait.PollUntilContextTimeout(context.Background(),
+	err := wait.PollUntilContextTimeout(ctx,
 		apiCallRetryInterval, kubeadmapi.GetActiveTimeouts().KubernetesAPICall.Duration,
-		true, func(_ context.Context) (bool, error) {
+		true, func(ctx context.Context) (bool, error) {
```

Comment:
this actually introduces a bug. other people have tried to do a similar change, but it's problematic.

what happen is that if a parent context is passed here the  lastError becomes "context deadline exceeded" once the timeout is reached. this later breaks some error type checking logic in kubeadm in various places. and the "context deadline exceeded" is not a very helpful message anyway.

we are not running full e2e suite on prs, but later this would have manifest as errors here https://testgrid.k8s.io/sig-cluster-lifecycle-kubeadm